zone/attack.cpp

```diff
@@ -5793,16 +5793,16 @@ void Mob::CommonOutgoingHitSuccess(Mob* defender, DamageHitInfo &hit, ExtraAttac
 	TryCriticalHit(defender, hit, opts);
 
 	hit.damage_done += hit.min_damage;
-	if (IsClient()) {
+	if (IsClient() || IsBot()) {
 		int extra = 0;
 		switch (hit.skill) {
-		case EQ::skills::SkillThrowing:
-		case EQ::skills::SkillArchery:
-			extra = CastToClient()->GetHeroicDEX() / 10;
-			break;
-		default:
-			extra = CastToClient()->GetHeroicSTR() / 10;
-			break;
+			case EQ::skills::SkillThrowing:
+			case EQ::skills::SkillArchery:
+				extra = GetHeroicDEX() / 10;
```

Comment:
Really need to be careful making these context changes

**mob.h**

```cpp
	inline virtual int32 GetHeroicSTR() const { return 0; }
	inline virtual int32 GetHeroicSTA() const { return 0; }
	inline virtual int32 GetHeroicDEX() const { return 0; }
	inline virtual int32 GetHeroicAGI() const { return 0; }
	inline virtual int32 GetHeroicINT() const { return 0; }
	inline virtual int32 GetHeroicWIS() const { return 0; }
	inline virtual int32 GetHeroicCHA() const { return 0; }
```

**client.h**

```cpp
	inline virtual int32 GetHeroicSTR() const { return itembonuses.HeroicSTR; }
	inline virtual int32 GetHeroicSTA() const { return itembonuses.HeroicSTA; }
	inline virtual int32 GetHeroicDEX() const { return itembonuses.HeroicDEX; }
	inline virtual int32 GetHeroicAGI() const { return itembonuses.HeroicAGI; }
	inline virtual int32 GetHeroicINT() const { return itembonuses.HeroicINT; }
	inline virtual int32 GetHeroicWIS() const { return itembonuses.HeroicWIS; }
	inline virtual int32 GetHeroicCHA() const { return itembonuses.HeroicCHA; }
```

Reply:
Let's make sure that clients are affected negatively by this

Reply:
Always best to be safe. and yes it works correctly (Polymorphism ftw!)
![image](https://user-images.githubusercontent.com/109764533/217116242-5dbb8eae-b06f-41a1-ba65-e761b26e4afd.png)

Reply:
All good - we just want to make sure things work we intend them to